runtime/src/iree/task/topology.h

```diff
@@ -190,12 +190,22 @@ iree_status_t iree_task_topology_initialize_from_logical_cpu_set(
 iree_status_t iree_task_topology_initialize_from_logical_cpu_set_string(
     iree_string_view_t cpu_id_set, iree_task_topology_t* out_topology);
 
+typedef enum iree_task_topology_performance_level_e {
+  // Selects all cores.
+  IREE_TASK_TOPOLOGY_PERFORMANCE_LEVEL_ANY = 0,
+  // Selects "E(fficiency)" cores that favor lower power/thermal load.
+  IREE_TASK_TOPOLOGY_PERFORMANCE_LEVEL_LOW,
+  // Selects "P(erformance)" cores that favor higher power/thermal load.
+  IREE_TASK_TOPOLOGY_PERFORMANCE_LEVEL_HIGH,
```

Comment:
So would we expect "ANY" to run faster than "HIGH", considering "ANY" uses performance _and_ efficiency cores? What sorts of use cases would want to use _only_ performance cores?

Reply:
added notes - it's all a tradeoff - if you allow efficiency cores you increase variability as what may take 50us on a performance core can take 500us on an efficiency one, and if work gets distributed across both and isn't nicely distributed by the compiler it can make things go real whacky (if the compiler does 4 tiles of 5ms of work on p cores and all 4 are grabbed by e cores then total latency may go from 5ms to 500ms). conversely, if the application is a game/user-interactive app and just running ML in the background they may want to force e cores because latency doesn't matter and it's better to keep the p cores for interactive tasks or not drain battery by keeping them powered on.